pkg/storage/replica_range_lease.go

```diff
@@ -127,7 +129,11 @@ func (p *pendingLeaseRequest) InitOrJoinRequest(
 		// Get the liveness for the next lease holder and set the epoch in the lease request.
 		liveness, err := repl.store.cfg.NodeLiveness.GetLiveness(nextLeaseHolder.NodeID)
 		if err != nil {
-			llChan <- roachpb.NewErrorf("couldn't request lease for %+v: %v", nextLeaseHolder, err)
+			llChan <- roachpb.NewError(&roachpb.LeaseRejectedError{
```

Comment:
Why is this just changing now? Can we add a test that would have demonstrated the need for this earlier?